src/java.desktop/unix/native/common/awt/fontpath.c

```diff
@@ -935,7 +935,7 @@ Java_sun_font_FontConfigManager_getFontConfig
         if (cacheDirs != NULL) {
             while ((cnt < max) && (cacheDir = (*FcStrListNext)(cacheDirs))) {
                 jstr = (*env)->NewStringUTF(env, (const char*)cacheDir);
-                JNU_CHECK_EXCEPTION(env);
+                JNU_CHECK_EXCEPTION_AND_CLEANUP(env, (*FcStrListDone)(cacheDirs));
```

Comment:
I think you do not need to create an additional macro here, just inline it and call "(*FcStrListDone)(cacheDirs);" directly. Something like:
```
if (IS_NULL(jstr) {
    (*FcStrListDone)(cacheDirs);
    return;
}
```
Note that the "IS_NULL" is used in this file after NewStringUTF. Any objections?

Reply:
Good catch! Elsewhere in this file IS_NULL is always used to check NewStringUTF, so the existing code was inconsistent in checking for a pending exception.

Checking for NULL and checking for a pending exception are logically equivalent as long as the passed in pointer is not NULL.